src/main/java/com/crud/tasks/trello/client/TrelloClient.java

```diff
@@ -0,0 +1,51 @@
+package com.crud.tasks.trello.client;
+
+import com.crud.tasks.domain.TrelloBoardDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+public class TrelloClient {
+
+    @Value("${trello.api.endpoint.prod}")
+    private String trelloApiEndpoint;
+
+    @Value("${trello.app.key}")
+    private String trelloAppKey;
+
+    @Value("${trello.app.token}")
+    private String trelloToken;
+
+    @Value("${trello.app.username}")
+    private String username;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    private URI buildUrl() {
+        URI url = UriComponentsBuilder.fromHttpUrl(trelloApiEndpoint + "/members/pan_woj/boards")
+                .queryParam("key", trelloAppKey)
+                .queryParam("token", trelloToken)
+                .queryParam("fields", "name,id").build().encode().toUri();
+
+        return url;
+    }
+
+    public Optional<List<TrelloBoardDto>> getTrelloBoards() {
+        TrelloBoardDto[] boardsResponse = restTemplate.getForObject(buildUrl(), TrelloBoardDto[].class);
+        List<TrelloBoardDto> boardsList = Arrays.asList(boardsResponse);
+        return Optional.ofNullable(boardsList);
+    }
```

Comment:
ta metoda jest do poprawienia
- Arrays.asList(boardsResponse); dla boardResponse == null, rzuci Ci exception'a
- Optional z kolekcji to anypattern. Jesli boardResponse bedzie nullem, twoja metoda pownnna zwrocic pusta liste (NIE nulla)